pkg/server/api/github.go

```diff
@@ -0,0 +1,123 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/bradleyfalzon/ghinstallation"
+	"github.com/robfig/cron/v3"
+	"github.com/shurcooL/githubv4"
+)
+
+const (
+	pageLength = 100
+)
+
+func (c *Api) getClient() *githubv4.Client {
+	tr := http.DefaultTransport
+
+	// We use a private key generated by GitHub along with the corresponding App ID and Installation ID
+	// to authenticate to GitHub
+	// https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/
+	// Ghinstallation creates a transport which we give to the githubv4 client for auth
+	itr, err := ghinstallation.NewKeyFromFile(
+		tr, c.conf.Github.AppId, c.conf.Github.InstallationId, c.conf.Github.PrivateKeyPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return githubv4.NewClient(&http.Client{Transport: itr})
+}
+
+func (c *Api) fetchUserMappings() (map[string]string, error) {
+	client := c.getClient()
+
+	var query struct {
```

Comment:
it could be helpful to move the GraphQL query here from the README actually (as a comment)